packages/components/src/date-time/time.js

```diff
@@ -23,347 +28,258 @@ import TimeZone from './timezone';
  */
 const TIMEZONELESS_FORMAT = 'YYYY-MM-DDTHH:mm:ss';
 
-class TimePicker extends Component {
-	constructor() {
-		super( ...arguments );
-		this.state = {
-			day: '',
-			month: '',
-			year: '',
-			hours: '',
-			minutes: '',
-			am: '',
-			date: null,
-		};
-		this.changeDate = this.changeDate.bind( this );
-		this.updateMonth = this.updateMonth.bind( this );
-		this.onChangeMonth = this.onChangeMonth.bind( this );
-		this.updateDay = this.updateDay.bind( this );
-		this.onChangeDay = this.onChangeDay.bind( this );
-		this.updateYear = this.updateYear.bind( this );
-		this.onChangeYear = this.onChangeYear.bind( this );
-		this.updateHours = this.updateHours.bind( this );
-		this.updateMinutes = this.updateMinutes.bind( this );
-		this.onChangeHours = this.onChangeHours.bind( this );
-		this.onChangeMinutes = this.onChangeMinutes.bind( this );
-		this.renderMonth = this.renderMonth.bind( this );
-		this.renderDay = this.renderDay.bind( this );
-		this.renderDayMonthFormat = this.renderDayMonthFormat.bind( this );
-	}
+/**
+ * <NumberInput>
+ * A shared component to parse, validate, and handle remounting of the underlying <input> element
+ *
+ * @param {Object} props Component props
+ * @param {string} props.as Render the component as specific element tag, defaults to "input"
+ * @param {number | string} props.value The default value of the component
+ * @param {Function} props.onUpdate Call back when blurred and validated
+ */
+function NumberInput( { as, value, onUpdate, ...props } ) {
+	function handleBlur( event ) {
+		const { target } = event;
 
-	componentDidMount() {
-		this.syncState( this.props );
-	}
+		if ( value === target.value ) {
+			return;
+		}
 
-	componentDidUpdate( prevProps ) {
-		const { currentTime, is12Hour } = this.props;
+		const parsedValue = parseInt( target.value, 10 );
+
+		// Run basic number validation on the input
 		if (
-			currentTime !== prevProps.currentTime ||
-			is12Hour !== prevProps.is12Hour
+			! isInteger( parsedValue ) ||
+			( typeof props.max !== 'undefined' && parsedValue > props.max ) ||
+			( typeof props.min !== 'undefined' && parsedValue < props.min )
 		) {
-			this.syncState( this.props );
+			// If validation failed, reset the value to the previous valid value
+			target.value = value;
+		} else {
+			// Otherwise, it's valid, call onUpdate
+			onUpdate( target.name, parsedValue );
 		}
 	}
+
+	return createElement( as || 'input', {
+		// Re-mount the input value to accept the latest value as the defaultValue
+		key: value,
```

Comment:
I noticed a small bug where the select options are closed when first clicking on the month field. Possibly related to the element being remounted in the DOM.

Reply:
Ahh! Right! Thx for noticing this!

Reply:
I believe this is caused by this part: https://github.com/WordPress/gutenberg/blob/e3d4e49553bbed7aaac341f6b613a3576c367568/packages/components/src/date-time/date.js#L29-L54. Yeah I agree it should be a separated issue.

Reply:
Found it! https://github.com/WordPress/gutenberg/issues/24206